projects/packages/publicize/src/rest-api/class-connections-controller.php

```diff
@@ -0,0 +1,434 @@
+	}
+
+	/**
+	 * Helper for retrieving Connections. Used by this endpoint and by
+	 * the Connection Test Result endpoint.
+	 *
+	 * @internal
+	 * @param bool $run_tests Whether to run the tests to check the health of the connections.
+	 * @return array
+	 */
+	protected static function get_connections( $run_tests = false ) {
+		global $publicize;
+
+		$items        = array();
+		$test_results = $run_tests ? self::get_connections_test_status() : array();
+
+		foreach ( (array) $publicize->get_services( 'connected' ) as $service_name => $connections ) {
+			foreach ( $connections as $connection ) {
+				$connection_meta = $publicize->get_connection_meta( $connection );
+				$connection_data = $connection_meta['connection_data'];
+				$connection_id   = $publicize->get_connection_id( $connection );
+				$username        = $publicize->get_username( $service_name, $connection );
+				$item            = array_merge(
+					array(
+						'connection_id'        => $connection_id,
+						'display_name'         => $publicize->get_display_name( $service_name, $connection ),
+						'external_handle'      => $username,
+						'external_id'          => $connection_meta['external_id'] ?? '',
+						'profile_link'         => $publicize->get_profile_link( $service_name, $connection ),
+						'profile_picture'      => $publicize->get_profile_picture( $connection ),
+						'service_label'        => Publicize::get_service_label( $service_name ),
+						'service_name'         => $service_name,
+						'shared'               => ! $connection_data['user_id'],
+						'user_id'              => (int) $connection_data['user_id'],
+
+						// Deprecated fields.
+						'id'                   => (string) $publicize->get_connection_unique_id( $connection ),
+						'username'             => $username,
+						'profile_display_name' => ! empty( $connection_meta['profile_display_name'] ) ? $connection_meta['profile_display_name'] : '',
+						// phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual -- We expect an integer, but do loose comparison below in case some other type is stored.
+						'global'               => 0 == $connection_data['user_id'],
+					),
+					$test_results[ $connection_id ] ?? array()
+				);
+				$items[]         = $item;
+			}
+		}
+
+		return $items;
+	}
+
+	/**
+	 * Get the connections test status.
+	 *
+	 * @return array
+	 */
+	protected static function get_connections_test_status() {
+		/**
+		 * Publicize instance.
+		 *
+		 * @var \Automattic\Jetpack\Publicize\Publicize $publicize
+		 */
+		global $publicize;
+
+		$test_results = $publicize->get_publicize_conns_test_results();
+
+		$test_results_map = array();
+		$mapping          = array(
+			'test_success'  => 'connectionTestPassed',
+			'test_message'  => 'connectionTestMessage',
+			'error_code'    => 'connectionTestErrorCode',
+			'can_refresh'   => 'userCanRefresh',
+			'refresh_text'  => 'refreshText',
+			'refresh_url'   => 'refreshURL',
+			'connection_id' => 'connectionID',
```

Comment:
I don't like these unused fields in this class.

Reply:
Fair enough, we can move them back to the subclass.